Serializer/Base64GzJson.php

```
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Doofinder\Feed\Serializer;
+
+use Magento\Framework\Serialize\Serializer\Json;
+
+/**
+ * Class for serializing data first to json string and then to base64 string.
+ *
+ * May be used for cases when json encoding results with a string,
+ * which contains characters, which are unacceptable by client.
+ */
+class Base64GzJson extends Json
+{
+    /**
+     * @inheritdoc
+     */
+    public function serialize($data)
+    {
+        return base64_encode(gzcompress(parent::serialize($data)));
+    }
+
+    /**
+     * Unserialize the given string with base64 and json.
+     * Falls back to the json-only decoding on failure.
+     *
+     * @param string $string
+     * @return string|int|float|bool|array|null
+     */
+
+    public function unserialize($string)
+    {
+        $decoded = base64_decode($string, true);
+        $decoded = base64_decode($string, true);
+        if ($decoded === false) {
+            return parent::unserialize($string);
+        }
+
+        // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged
+        $uncompressed = @gzuncompress($decoded);
```

Is it necessary to silence this error? Just asking for curiosity.

---

It isn't strictly necessary, but since `gzuncompress` does not throw an exception and we only need to check whether the data is anything other than `false`, the warning is irrelevant.

That said, I can remove the suppression.

---

In that case I think if it can be safely removed, then it's better to remove it :smile: (Although there are many excepctions to this rule, of course :raised_hands: )